batch/batch/batch.py

```diff
@@ -655,40 +665,16 @@ def is_successful(self):
             exit_codes = [None for _ in tasks]
         if durations is None:
             durations = [None for _ in tasks]
-        n_updated = await db.jobs.update_record(*self.id,
-                                                compare_items={'state': self._state},
-                                                durations=json.dumps(durations),
-                                                exit_codes=json.dumps(exit_codes),
-                                                state=new_state)
-
-        if n_updated == 0:
-            log.info(f'could not update job {self.id} due to db not matching expected state')
-            raise JobStateWriteFailure()
-
-        self.exit_codes = exit_codes
-        self.durations = durations
-
-        if self._state != new_state:
-            log.info('job {} changed state: {} -> {}'.format(
-                self.id,
-                self._state,
-                new_state))
-
-        self._state = new_state
-
-        await app['pod_throttler'].delete_pod(self)
-        await self._delete_pvc()
-        await self.notify_children(new_state)
-
-        log.info('job {} complete with state {}, exit_codes {}'.format(self.id, self._state, self.exit_codes))
-
+        await self._delete_k8s_resources()
```

Comment:
Shouldn't we set the state before deleting resources?

Reply:
done.